dns.c

```diff
@@ -46,18 +48,33 @@ int type;
  errno = 0;
  if (!stralloc_copy(&glue,domain)) return DNS_MEM;
  if (!stralloc_0(&glue)) return DNS_MEM;
- responselen = lookup(glue.s,C_IN,type,response.buf,sizeof(response));
+ if (!responsebuflen)
+  if (response.buf = (unsigned char *)alloc(PACKETSZ+1))
```

Comment:
Will likely warn because it needs an extra pair of (). And as this is considered "new" code it should use malloc() directly and doesn't need the cast then.

Reply:
Fixed in #201.